### PR TITLE
cs/refactor/bulk break up zip

### PR DESCRIPTION
**Pull request by chancesmith**
## overall
Zip file function does too many things. Plus, we don't have any tests surrounding files we expect fields we require.

## changes made
- breakup parts of creating zip file
- test what files we expect in zip
- test fields we expect to be required

## screenshot
Screen Shot 2023-01-08 at 6 54 46 AM

### CODE REVIEW

File: csv-bulk-signature/src/main.ts [hunk 1]
```diff
@@ -222,6 +220,16 @@ function skipFirstPlaceholderRow(contact: Contact) {
   return contact["Brand*"] !== "use dropdown to select a brand";
 }
 
+export function filterFilesForZip(file: string) {
+  return file.endsWith(".htm") || file.endsWith(".txt");
+}
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:225]: extract for testing and outside of the zip file creation

File: csv-bulk-signature/src/main.ts [hunk 2]
```diff
@@ -222,6 +220,16 @@ function skipFirstPlaceholderRow(contact: Contact) {
...
+
+export const checkRequiredFields = (row: Contact) =>
+  !!row["Brand*"]?.length &&
+  !!row["Full Name*"]?.length &&
+  !!row["Title*"]?.length &&
+  !!row["Office Phone*"]?.length;
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:231]: add optional chaining for if one required field doesn't exist and we can't get the length of it

File: csv-bulk-signature/src/main.ts [hunk 3]
```diff
@@ -238,12 +246,6 @@ async function main() {
   const contacts = await getCSVRows(CSV_FILE);
   await checkHeadersToBeSame(contacts[0]);
 
-  const checkRequiredFields = (row: Contact) =>
-    !!row["Brand*"].length &&
-    !!row["Full Name*"].length &&
-    !!row["Title*"].length &&
-    !!row["Office Phone*"].length;
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:245]: extract for testing

File: csv-bulk-signature/src/main.ts [hunk 4]
```diff
@@ -255,8 +257,12 @@ async function main() {
...
 
-  // 6. zip up files
-  await zipUpFile();
+  // 6. gather files for zip
+  const signatureFolder = await fs.readdir(SIGNATURES_PATH);
+  const files = signatureFolder.filter(filterFilesForZip);
+
+  // 7. zip up files
+  await zipUpFiles(files);
   await checkZipIsNotEmpty(ZIP_FILE);
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:266]: gather the files as one step to trim out the multiple test that were in zipUpFiles

File: csv-bulk-signature/src/main.ts [hunk 5]
```diff
@@ -148,12 +148,7 @@ export function getFileName(contact: Contact) {
   return lowerCaseFullName;
 }
 
-async function zipUpFile() {
-  const signatureFolder = await fs.readdir(SIGNATURES_PATH);
-  const files = signatureFolder.filter(
-    (file) => file.endsWith(".htm") || file.endsWith(".txt")
-  );
-
+export async function createZipFile(files: string[]) {
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:151]: create new function that handles the zip file creation alone

File: csv-bulk-signature/src/main.ts [hunk 6]
```diff
@@ -162,9 +157,12 @@ async function zipUpFile() {
...
 
-  try {
-    const buffer = await zip.generateAsync({ type: "nodebuffer" });
+  return zip.generateAsync({ type: "nodebuffer" });
+}
 
+export async function zipUpFiles(files: string[]) {
+  try {
+    const buffer = await createZipFile(files);
     await fs.writeFile(ZIP_FILE, buffer);
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:166]: now `zipUpFiles` takes in list of files as argument and calls another function to gather the zip buffer, then we just write to the file system.

File: csv-bulk-signature/src/main.test.ts [hunk 7]
```diff
@@ -52,3 +57,64 @@ describe("getFileName", () => {
...
+
+  const result = files.filter(filterFilesForZip);
+
+  expect(result).toStrictEqual([
+    "acurl.htm",
+    "anitti.htm",
+    "ndodge.htm",
+    "_statusReport.txt",
+  ]);
+});
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.test.ts:78]: test that we expect certain files in the zip (output)

File: csv-bulk-signature/src/main.test.ts [hunk 8]
```diff
@@ -52,3 +57,64 @@ describe("getFileName", () => {
...
+
+    expect(
+      checkRequiredFields({
+        ...contactGenerator(),
+        // @ts-ignore - we're assuming Contact had missing data in runtime
+        "Title*": undefined,
+      })
+    ).toBe(false);
+  });
+});
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.test.ts:120]: test to expect required folds from csv (input)

File: csv-bulk-signature/src/main.test.ts [hunk 9]
```diff
@@ -34,7 +39,7 @@ describe("getFullNameFileName", () => {
 
 // test getFileName
 describe("getFileName", () => {
-  it("should return the correct file name", () => {
+  it("should return file name of first initial + last name", () => {
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.test.ts:42]: better explain test name

**Review by adamnitti — APPROVED** [review]

**Pull request merged**